src/test/java/com/artipie/npm/proxy/http/DownloadAssetSliceTest.java

```diff
@@ -0,0 +1,174 @@
+import org.junit.jupiter.params.provider.ValueSource;
+
+/**
+ * Test cases for {@link DownloadAssetSlice}.
+ * @since 0.9
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+@SuppressWarnings({"PMD.AvoidUsingHardCodedIP", "PMD.AvoidDuplicateLiterals"})
+final class DownloadAssetSliceTest {
+
+    /**
+     * Vertx.
+     */
+    private Vertx vertx;
+
+    /**
+     * NPM Proxy.
+     */
+    private NpmProxy npm;
+
+    /**
+     * Server port.
+     */
+    private int port;
+
+    /**
+     * TgzArchive path.
+     */
+    private String tgzpath;
+
+    @BeforeEach
+    void setUp() throws InterruptedException, ExecutionException, IOException {
+        this.vertx = Vertx.vertx();
+        final Storage storage = new InMemoryStorage();
+        this.tgzpath =
+            "@hello/simple-npm-project/-/@hello/simple-npm-project-1.0.1.tgz";
+        storage.save(
+            new Key.From(this.tgzpath),
+            new Content.From(
+                new TestResource(
+                    String.format("storage/%s", this.tgzpath)
+                ).asBytes()
+            )
+        ).get();
+        storage.save(
+            new Key.From(
+                String.format("%s.meta", this.tgzpath)
+            ),
+            new Content.From(
+                Json.createObjectBuilder()
+                    .add("last-modified", "2020-05-13T16:30:30+01:00")
+                    .build()
+                    .toString()
+                    .getBytes()
+            )
+        ).get();
+        this.port = new NextSafeAvailablePort().value();
+        final YamlMapping yaml = Yaml.createYamlMappingBuilder()
+            .add(
+                "remote",
+                Yaml.createYamlMappingBuilder()
+                    .add(
+                        "url",
+                        String.format("http://127.0.0.1:%d", this.port)
+                    )
+                    .build()
+            ).build();
+        final  NpmProxyConfig config = new NpmProxyConfig(yaml);
+        this.npm = new NpmProxy(config, this.vertx, storage);
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"", "ctx"})
+    void downloadMetaWorks(final String pathprefix) {
+        final AssetPath path = new AssetPath(pathprefix);
+        try (
+            VertxSliceServer server = new VertxSliceServer(
+                this.vertx,
+                new DownloadAssetSlice(this.npm, path),
+                this.port
+            )
+        ) {
+            server.start();
+            String baseurl = String.format(
+                "http://127.0.0.1:%d",
+                this.port
+            );
+            if (!pathprefix.isEmpty()) {
+                baseurl = String.format(
+                    "%s/%s",
+                    baseurl,
+                    pathprefix
+                );
+            }
+            final String url = String.format(
+                "%s/%s",
+                baseurl,
+                this.tgzpath
+            );
```

Comment:
@baudoliver7 check for emptiness in this case by specifying `ctx/` instead of `ctx`

Reply:
@genryxy I afraid we can do that because of :
```suggestion
final AssetPath path = new AssetPath(pathprefix);
```
We should not have slash in `pathprefix.`

Reply:
@baudoliver7 yeah, I missed this statement. `prefix.replaceAll("/$")` will solve this problem.

Reply:
@genryxy I get it now. But I did it in another manner :)